File: other/deepseek/ReverseString2.js

const reverseString = (nums) => {

	let first = 0;
	let second = nums.length - 1;

	while (first < second){

		let temp = nums[first];
		nums[first] = nums[second];
		nums[second] = temp;
		first ++;
		second --;
	}

	return nums;
}

let nums = ['h','e','l','l','o'];

for (let m of reverseString(nums)){

	console.log(m);
}

File: other/deepseek/ReverseString2.java
import java.util.Arrays;

// Bir karakter dizisini (char[] s) yerinde (in-place) ters çevirin. Ekstra bellek kullanmayın.

public class ReverseString2{

    private static void reverseString(char[] s) {

        if (s == null || s.length <= 1) return;

        int left = 0;
        int right = s.length - 1;

        while (right > left) {
            char temp = s[left];
            s[left++] = s[right];
            s[right--] = temp;
        }
    }

    public static void main(String[] args){

        char[] s = {'h','e','l','l','o'};

        reverseString(s);
        System.out.println(Arrays.toString(s));
    }
}